### ISSUE TITLE
Indentation bug in TensorFlow main expert tutorial

### ISSUE THREAD

**Issue by doughywilson** [tensorflow/tensorflow#57536]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Documentation Bug

### Source

binary

### Tensorflow Version

2.8

### Custom Code

Yes

### Current Behaviour?

```shell
There is an indentation bug in the main Tensorflow expert colab tutorial located here: https://colab.research.google.com/github/tensorflow/docs/blob/master/site/en/tutorials/quickstart/advanced.ipynb. The code does not run because there is an indentation problem.

@tf.function
def train_step(images, labels):
  with tf.GradientTape() as tape:
    # training=True is only needed if there are layers with different
    # behavior during training versus inference (e.g. Dropout).
    predictions = model(images, training=True)
    loss = loss_object(labels, predictions)
  gradients = tape.gradient(loss, model.trainable_variables)  #PROBLEM!!! NEEDS INDENTATION!
  optimizer.apply_gradients(zip(gradients, model.trainable_variables))

  train_loss(loss)
  train_accuracy(labels, predictions)
```

### Standalone code to reproduce the issue

```shell
Run the colab all the way through, and you will see that it chokes on the last step.

https://colab.research.google.com/github/tensorflow/docs/blob/master/site/en/tutorials/quickstart/advanced.ipynb
```

### Relevant log output

_No response_</details>

**Comment by mohantym** [comment]
Hi @doughywilson !
It was running fine in Colab with the existing tutorial, But I raised a [PR](https://github.com/tensorflow/docs/pull/2120) to fix the indentation issue.       
Thank you!

**Comment by MarkDaoust** [comment]
> #PROBLEM!!! NEEDS INDENTATION!

It doesn't actually. Why do you think it does?
Closing the tape is "stop writing new gradient ops to the tape", not "throw out the tape".

**Comment by 8bitmp3** [comment]
Closing the issue for now, as it appears to be a non-issue at the moment.

> It was running fine in Colab with the existing tutorial

👍

**Comment by mohantym** [comment]
@MarkDaoust @8bitmp3 !
Thanks for the clarification.
